scipy/linalg/_basic.py

```diff
@@ -1435,14 +1436,15 @@ def pinv(a, atol=None, rtol=None, return_rank=False, check_finite=True,
     t = u.dtype.char.lower()
     maxS = np.max(s)
 
-    if rcond or cond:
+    if rcond is not _NoValue or cond is not _NoValue:
         warn('Use of the "cond" and "rcond" keywords are deprecated and '
-             'will be removed in future versions of SciPy. Use "atol" and '
+             'will be removed in SciPy 1.14.0. Use "atol" and '
              '"rtol" keywords instead', DeprecationWarning, stacklevel=2)
```

Comment:
Given the complexity of the whole tol/atol/rtol operation, and the fact that we were warning on this consistently already, I don't think we _need_ the full two releases here. That said, #18488 promises removal by 1.14 as well, so I'm fine with committing 1.14 in this PR. Basically I wanted to leave the door open for us to do it earlier if we feel the trade-off is worth it.